src/transforms/reduce/config.rs

```diff
@@ -100,6 +101,121 @@ pub struct ReduceConfig {
     pub starts_when: Option<AnyCondition>,
 }
 
+/// Incoming Event Conditional
+#[configurable_component]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct IncomingEventCondition {
+    /// Condition to apply
+    #[serde(flatten)]
+    pub condition: AnyCondition,
+}
+
+/// Merged Event Condition
+#[configurable_component]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct MergedEventCondition {
+    /// Condition to apply
+    #[serde(flatten)]
+    pub condition: AnyCondition,
+}
+
+/// Condition mode for ends_with
+#[configurable_component(no_deser)]
+#[derive(Clone, Debug)]
+#[serde(tag = "apply_to")]
+#[serde(deny_unknown_fields)]
+#[allow(clippy::large_enum_variant)] // just used for configuration
+pub enum ApplyTo {
+    /// Run condition on the incoming event
+    #[serde(rename = "incoming_event")]
+    IncomingEvent(IncomingEventCondition),
+
+    /// Run condition on the merged event
+    #[serde(rename = "merged_event")]
+    MergedEvent(MergedEventCondition),
+}
+
+mod deser {
+    use super::*;
+
+    #[derive(Deserialize)]
+    #[serde(tag = "apply_to")]
+    enum ApplyToTagged {
+        #[serde(rename = "incoming_event")]
+        IncomingEvent(IncomingEventCondition),
+
+        #[serde(rename = "merged_event")]
+        MergedEvent(MergedEventCondition),
+    }
+
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum ApplyToDe {
+        Tagged(ApplyToTagged),
+
+        // Note: this must be last as serde attempts variants in order
+        Untagged(AnyCondition),
+    }
+
+    impl<'de> Deserialize<'de> for ApplyTo {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: serde::Deserializer<'de>,
+        {
+            Ok(match ApplyToDe::deserialize(deserializer)? {
+                ApplyToDe::Tagged(ApplyToTagged::IncomingEvent(config)) => {
+                    ApplyTo::IncomingEvent(config)
+                }
+                ApplyToDe::Tagged(ApplyToTagged::MergedEvent(config)) => {
+                    ApplyTo::MergedEvent(config)
+                }
+                ApplyToDe::Untagged(config) => {
+                    ApplyTo::IncomingEvent(IncomingEventCondition { condition: config })
+                }
+            })
+        }
+    }
+
+    #[cfg(all(test))]
```

Comment:
```suggestion
    #[cfg(test)]
```

Clippy should complain about this.